sechub-commons-model/src/main/java/com/mercedesbenz/sechub/commons/model/SecHubRemoteCredentialSupport.java

```diff
@@ -0,0 +1,65 @@
+package com.mercedesbenz.sechub.commons.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SecHubRemoteCredentialSupport {
+    private final String PDS_PREPARE_REMOTE_CREDENTIAL_LIST = "PDS_PREPARE_REMOTE_CREDENTIAL_LIST";
+    private static final Logger LOG = LoggerFactory.getLogger(SecHubRemoteCredentialSupport.class);
+
+    public List<SecHubRemoteCredentialContainer> resolveCredentialsForLocation(SecHubRemoteCredentialConfiguration configuration, String location) {
+        return matchRemoteCredentialsFromLocationConfiguration(configuration, location, null);
+    }
+
+    public List<SecHubRemoteCredentialContainer> resolveCredentialsForLocation(SecHubRemoteCredentialConfiguration configuration, String location,
+            String type) {
+        TypeFilter.AcceptTypeFilter acceptTypeFilter = new TypeFilter.AcceptTypeFilter(type);
+        return matchRemoteCredentialsFromLocationConfiguration(configuration, location, acceptTypeFilter);
+    }
+
+    public List<SecHubRemoteCredentialContainer> matchRemoteCredentialsFromLocationConfiguration(SecHubRemoteCredentialConfiguration configuration,
+            String location, TypeFilter filter) {
+        if (filter == null) {
+            filter = TypeFilter.ACCEPT_ALL;
+        }
+        List<SecHubRemoteCredentialContainer> matchedCredentials = new ArrayList<>();
+
+        if (location == null) {
+            LOG.debug("Could not match credentials as configured location was null.");
+            return matchedCredentials;
+        }
+
+        for (SecHubRemoteCredentialContainer credentialContainer : configuration.getCredentials()) {
+            String stringPattern = credentialContainer.getRemotePattern();
+            String type = credentialContainer.getType();
+
+            if (stringPattern == null) {
+                LOG.debug("Could not match credentials as configured remote pattern was null.");
+                continue;
+            }
+
+            Pattern pattern = Pattern.compile(stringPattern);
```

Comment:
Pattern compile can throw an exception.

Please do following:

1. write a unit test with some data containing illegal regular expressions.
   expect no exception here, but the part with the wrong configuration lead to a missing entry. I would expect here a warn log entry
2. change the implementation